metaflow/plugins/argo/argo_workflows.py

```diff
@@ -574,6 +574,11 @@ def _compile_workflow_template(self):
             "metaflow/user": "argo-workflows",
             "metaflow/flow_name": self.flow.name,
         }
+
+        for name, parameter in self.parameters.items():
+            if parameter["is_required"]:
+                annotations[f"metaflow/parameter-required-{name}"] = "true"
```

Comment:
one thing to note is that there does not seem to be a max length to parameter names, but there is a hard limit on what is allowed in Kubernetes annotations: 63 characters for the 'name' segment (the key without prefix). The size constraint for annotation value is much more lenient, though not as well documented. Something like 
```
annotations["metaflow/required-parameters"] = "paramA,paramB, paramC,..."
```
should presumably fit in the annotations value though?

Reply:
Thanks for the feedback!  I'm going with your suggestion now - I had to switch strategies anyway because `/` is an allowed character in metaflow parameter names but not in annotations.